tests/unit/test_charm.py

```diff
@@ -3,6 +3,151 @@
 #
 # Learn more about testing at: https://juju.is/docs/sdk/testing
 
+import ops
+import pytest
+from ops import testing
+from unittest.mock import patch, PropertyMock
+from src.charm import CassandraCharm
+from src.core.state import PEER_RELATION
+
+
+def test_start_maintenance_status_when_starting():
+    """Charm enters MaintenanceStatus when Cassandra is not yet healthy."""
+    ctx = testing.Context(CassandraCharm)
+    relation = testing.PeerRelation(id=1, endpoint=PEER_RELATION)
+    state_in = testing.State(leader=True, relations={relation})
+
+    with (
+        patch("managers.config.ConfigManager.render_env"),
+        patch("workload.CassandraWorkload.start"),
```

Comment:
`restart` is actually used here. Until corrected in charm codebase, let's just pull it for now. **This also applies to other tests.**
```suggestion
        patch("workload.CassandraWorkload.restart"),
```